build.sbt

```diff
@@ -367,7 +367,7 @@ lazy val main = project
   .settings(
     packageName in Docker := "gem",
     dockerBaseImage       := "openjdk:8u141",
-    dockerExposedPorts    := List(6666),
+    dockerExposedPorts    := List(9090, 9091),
```

Comment:
Do these also need to be opened in the host's firewall somehow?

Reply:
I don't think the VMs have firewalls running since they're internal. But eventually yes.

Reply:
right, can docker do it automagically? Does it run as root?

Reply:
Containers can run services on any port because they're sandboxed. These ports can be seen by other containers on the same machine if they join the same virtual network. This is how Gem can see Postgres (but nobody else can).

To see a container's port *externally* it must be mapped to a unique port on the host. We expose gem's ports 9090 and 9091 this way, using the same port numbers on the host for simplicity.

Docker itself runs as `docker` (not `root`) so I don't think you can map a container port to a host port < 1024 without special permissions. I don't think it has anything to say about a firewall running on the host.

Reply:
nice, container isolation cannot be bad while letting them talk to each other
BTW i always like to use the standard ports like 80 and 443 for the web part. What I've found the easiest is to get ITS to forward those ports to your app in the local firewall

Reply:
I'm not sure docker can forward those ports but yeah you're right. Might as well if I can. I'll try it.